### PR TITLE
chore(dispatcher): refactor out training code

### PR DESCRIPTION
**Pull request by sauyon**
Refactor to make this code more reasonable.

### CODE REVIEW

**Review by aarnphm — APPROVED** [review]
Only if github has a better diff view

**Comment by sauyon** [comment]
This is just a nightmare diff in general :P

File: src/bentoml/_internal/marshal/dispatcher.py [hunk 1]
```diff
@@ -168,52 +168,18 @@ async def _func(data: t.Any) -> t.Any:
...
-                # call
-                self._sema.acquire()
-                inputs_info = tuple(self._queue.pop() for _ in range(n_call_out))
-                self._loop.create_task(self.outbound_call(inputs_info))
-            except asyncio.CancelledError:
-                return
-            except Exception as e:  # pylint: disable=broad-except
-                logger.error(traceback.format_exc(), exc_info=e)
+    async def train_optimizer(
+        self, num_required_reqs: int, num_reqs_to_train: int, batch_size: int, wait: 0,
```
Comment (aarnphm) [inline src/bentoml/_internal/marshal/dispatcher.py:172]:
```suggestion
        self, num_required_reqs: int, num_reqs_to_train: int, batch_size: int, wait: int = 0,
```
I think this is what you mean?

Comment (sauyon) [reply src/bentoml/_internal/marshal/dispatcher.py:172]: Oh, no, I just forgot to remove that argument.

**Review by aarnphm — APPROVED** [review]
LGTM. cc @bojiang

**Pull request merged**